components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java

```diff
@@ -108,16 +108,16 @@ public static SubscriptionThrottlePolicyDTO fromSubscriptionThrottlePolicyToDTO(
                     CommonThrottleMappingUtil.fromQuotaPolicyToDTO(subscriptionPolicy.getDefaultQuotaPolicy()));
         }
         policyDTO.setType(SUBSCRIPTION_THROTTLE_POLICY_TYPE);
+        MonetizationInfoDTO monetizationInfoDTO = new MonetizationInfoDTO();
         if (APIConstants.COMMERCIAL_TIER_PLAN.equals(subscriptionPolicy.getBillingPlan())) {
-            MonetizationInfoDTO monetizationInfoDTO = new MonetizationInfoDTO();
             if (APIConstants.Monetization.FIXED_RATE.equalsIgnoreCase(subscriptionPolicy.getMonetizationPlan())) {
                 monetizationInfoDTO.setMonetizationPlan(MonetizationInfoDTO.MonetizationPlanEnum.FIXEDRATE);
             } else {
                 monetizationInfoDTO.setMonetizationPlan(MonetizationInfoDTO.MonetizationPlanEnum.DYNAMICRATE);
             }
             monetizationInfoDTO.setProperties(subscriptionPolicy.getMonetizationPlanProperties());
-            policyDTO.setMonetization(monetizationInfoDTO);
         }
+        policyDTO.setMonetization(monetizationInfoDTO);
```

Comment:
Do we need to set this , if the subscription policy is not a commercial one?

Reply:
Yes, the `monetizationPlan` is marked as a required attribute and it is an enum - [1]. Currently, it returns null if `COMMERCIAL_TIER_PLAN` is not selected - [2] and throws NPE when editing subscription policy [3]. Earlier this issue was not there as if the `monetizationPlan` is null then UI sent `FIXEDRATE`. This logic is changed in a recent fix.

So in order to fix it, either we should mark it as a not-required value or set a default value. Then shall we mark it as a not-required attribute. WDYT?

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml#L3522 
[2] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java#L111
[3] https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx#L258

Reply:
+1

Reply:
Fixed in db16b57